### PR TITLE
 C++17 cleanup: traits replace std::enable_if<>::type with std::enable_if_t

### PR DESCRIPTION
**Pull request by karthikeyann**

### CODE REVIEW

**Review by bdice — APPROVED** [review]
👍

**Comment by bdice** [comment]
We have a macro `CUDF_ENABLE_IF` as well. I’m not sure if we should try to standardize that.

**Comment by karthikeyann** [comment]
> We have a macro `CUDF_ENABLE_IF` as well. I’m not sure if we should try to standardize that (or remove the macro).

Likely most of its usage will be replaced with `if constexpr`, `std::invocable` and expression SFINAE. If all of them are removed, that macro can be removed.

**Review by davidwendt — APPROVED** [review]

**Review by robertmaynard — APPROVED** [review]

File: cpp/include/cudf/utilities/span.hpp [hunk 1]
```diff
@@ -159,9 +159,9 @@ struct host_span : public cudf::detail::span_base<T, Extent, host_span<T, Extent
   // Copy construction to support const conversion
   template <typename OtherT,
             std::size_t OtherExtent,
-            typename std::enable_if<(Extent == OtherExtent || Extent == dynamic_extent) &&
-                                      std::is_convertible_v<OtherT (*)[], T (*)[]>,
-                                    void>::type* = nullptr>
+            typename std::enable_if_t<(Extent == OtherExtent || Extent == dynamic_extent) &&
+                                        std::is_convertible_v<OtherT (*)[], T (*)[]>,
+                                      void>* = nullptr>
```
Comment (jrhemstad) [inline cpp/include/cudf/utilities/span.hpp:164]: Might as well go all the way. Or, can just use `CUDF_ENABLE_IF` macro. 

```suggestion
            std::enable_if_t<(Extent == OtherExtent || Extent == dynamic_extent) &&
                                        std::is_convertible_v<OtherT (*)[], T (*)[]>>* = nullptr>
```

**Review by ttnghia — APPROVED** [review]

File: cpp/include/cudf/utilities/span.hpp (hunk 1 above)
Comment (karthikeyann) [reply cpp/include/cudf/utilities/span.hpp]: updated.

**Comment by karthikeyann** [comment]
rerun tests

**Comment by karthikeyann** [comment]
@gpucibot merge

**Pull request merged**